craft_parts/plugins/plugins.py

```diff
@@ -116,10 +116,19 @@ def register(plugins: Dict[str, PluginType]) -> None:
     _PLUGINS.update(plugins)
 
 
-def unregister_all() -> None:
-    """Unregister all user-registered plugins."""
+def unregister(*plugins: str) -> None:
+    """Unregister plugins by name."""
+    for plugin in plugins:
+        _PLUGINS.pop(plugin, None)
+
+
+def unregister_all(*, unregister_built_in: bool = False) -> None:
```

Comment:
With the `unregister` suggested by @lengau the `unregister_built_in` flag becomes unnecessary.